src/System.CodeDom/src/Microsoft/VisualBasic/VBCodeGenerator.cs

```diff
@@ -0,0 +1,2931 @@
+                    // If current character is a high surrogate and the following 
+                    // character is a low surrogate, don't break them. 
+                    // Otherwise when we write the string to a file, we might lose 
+                    // the characters.
+                    // 
+                    if (char.IsHighSurrogate(value[i])
+                        && (i < value.Length - 1)
+                        && char.IsLowSurrogate(value[i + 1]))
+                    {
+                        b.Append(value[++i]);
+                    }
+
+                    if (fInDoubleQuotes)
+                        b.Append("\"");
+                    fInDoubleQuotes = true;
+
+                    b.Append("& _ ");
+                    b.Append(Environment.NewLine);
+                    b.Append(indentObj.IndentationString);
+                    b.Append('\"');
+                }
+                ++i;
+            }
+
+            if (fInDoubleQuotes)
+                b.Append("\"");
+
+            return b.ToString();
+        }
+
+        private static void AppendEscapedChar(StringBuilder b, char value)
+        {
+            b.Append("&Global.Microsoft.VisualBasic.ChrW(");
+            b.Append(((int)value).ToString(CultureInfo.InvariantCulture));
+            b.Append(")");
+        }
+
+        protected override void ProcessCompilerOutputLine(CompilerResults results, string line)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        protected override string CmdArgsFromParameters(CompilerParameters options)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        protected override void OutputAttributeArgument(CodeAttributeArgument arg)
+        {
+            if (arg.Name != null && arg.Name.Length > 0)
+            {
+                OutputIdentifier(arg.Name);
+                Output.Write(":=");
+            }
+            ((ICodeGenerator)this).GenerateCodeFromExpression(arg.Value, ((ExposedTabStringIndentedTextWriter)Output).InnerWriter, Options);
+        }
+
+        private void OutputAttributes(CodeAttributeDeclarationCollection attributes, bool inLine)
+        {
+            OutputAttributes(attributes, inLine, null, false);
+        }
+
+        private void OutputAttributes(CodeAttributeDeclarationCollection attributes, bool inLine, string prefix, bool closingLine)
+        {
+            if (attributes.Count == 0) return;
+            bool firstAttr = true;
+            GenerateAttributeDeclarationsStart(attributes);
+            foreach (CodeAttributeDeclaration current in attributes)
+            {
+                if (firstAttr)
+                {
+                    firstAttr = false;
+                }
+                else
+                {
+                    Output.Write(", ");
+                    if (!inLine)
+                    {
+                        ContinueOnNewLine("");
+                        Output.Write(" ");
```

Comment:
Changed all relevant single-character strings to be chars, both with StringBuilder.Append and TextWriter.Output.Write{Line}